### PR TITLE
enable building rh9 rpm with shared openssl

### PR DESCRIPTION
**Pull request by qrkourier**

### CODE REVIEW

**Comment by qrkourier** [comment]
Clarification: this change makes it possible for us to test the OpenSSL build using the CPack builder machinery. This does not change the release build to OpenSSL from static Mbed-TLS.

File: .github/actions/openziti-tunnel-build-action/redhat-9/entrypoint.sh [hunk 1]
```diff
@@ -45,6 +45,7 @@ done
         --preset "${cmake_preset}" \
         -DCMAKE_BUILD_TYPE="${cmake_config}" \
         -DBUILD_DIST_PACKAGES=ON \
+        "${TLSUV_TLSLIB:+-DTLSUV_TLSLIB=${TLSUV_TLSLIB}}" \
```
Comment (scareything) [inline .github/actions/openziti-tunnel-build-action/redhat-9/entrypoint.sh:48]: Why do this for redhat-9 only? If this is just a quick trick for us then maybe a comment here to explain the purpose.

Is there any value in doing it for other distros (keeping in mind that just setting TLSUV_TLSLIB to openssl is only sufficient on distros that already have libssl3 - other distros will also need "openssl" to be added to vcpkg.json). 

There is one downside to this and I'm not sure how to handle it. "mbedtls" is currently defined as a dependency in vcpkg.json, which means that vcpkg will download and build mbedtls whether or not the build actually uses it (which is determined by TLSUV_TLSLIB).

**Review by scareything — APPROVED** [review]

File: .github/actions/openziti-tunnel-build-action/redhat-9/Dockerfile [hunk 2]
```diff
@@ -1,8 +1,10 @@
 ARG CMAKE_VERSION="3.26.3"
+ARG VCPKG_VERSION="2024.01.12"
```
[the remark this answers is not in the document]
Comment (qrkourier) [reply .github/actions/openziti-tunnel-build-action/redhat-9/Dockerfile:2]: I see what you mean. I'd like to converge on a pattern we can use for the CPack cross-builder images as well as the generic ziti-builder image (cross-builder for everything else that uses the C-SDK that's not a ZET Linux package). I'll keep the trunk checkout idea in mind.

File: .github/actions/openziti-tunnel-build-action/redhat-9/entrypoint.sh (hunk 1 above)
Comment (qrkourier) [reply .github/actions/openziti-tunnel-build-action/redhat-9/entrypoint.sh:48]: I extended the TLS lib override to RedHat9 so I could try to reproduce the enrollment problem in https://github.com/openziti/ziti-tunnel-sdk-c/issues/517#issuecomment-1947494141.

I see what you mean about building Mbed-TLS unnecessarily. I see how to temporarily remove it from vcpkg's deps list, or to replace it with "openssl" for building it statically for distros with libssl v1.

**Pull request merged**